Story/918

crwr45 opened the pull request.

Commit:
[IMP] Adding facility to block work on a picking until all its dependencies are completed.

udes_core:
Added stock.picking.u_pending field
 - computed, shows if picking has waiting moves.
 - If pending, a picking cannot be validated.
 - u_pending only included in api returns if picking.type.u_handle_partials is False.

udes_closed:
 - Added check to pallet-move controller to prevent working on pending pickings

warehouse_config
Added stock.picking.type.u_handle_partials
 - Controls use of u_pending.
 - If False, pending pickings of that type cannot be worked on.

Commit:
[FIX] Removing circular dependency and unnecessary decorator.

Commit:
[IMP] Adding Python Unit Tests for handle_partials functionality.

Commit:
[FIX] Correcting field usage for dependencies

Previously had code in udes_core that was explicitly using a
field defined in warehouse_config, but udes_core does not have a
dependency on warehouse_config.
As a result the code has now been split so as to not create undeclared
dependencies, and the tests have gone into udes_api, which requires
both udes_core and warehouse_config to work.

ws-mps committed:
Merge branch 'develop' into story/918

ws-mps commented on models/stock_picking.py:
```diff
@@ -642,6 +675,7 @@ def _prepare_info(self, priorities=None, fields_to_fetch=None):
             - location_dest_id: int
             - picking_type_id: int
             - move_lines: [{stock.move}]
+            - u_partial: boolean (only if picking type does not handle partials)
```
I think this should be u_pending

ws-mps committed:
[918] Fix for canceled previous picks

When one of the previous picks has been canceled, the state of
the stock move is still waiting or partial, so instead of
checking the state of the goods-out move is best to check the
state of the origin moves of those.

ws-mps committed:
[918] Fix typo and use u_prev_picking_ids

Instead of using current move states, use the state of the
previous pickings since it will endup with shorter lists.

nocturnalastro commented on models/stock_picking.py:
```diff
@@ -77,17 +77,16 @@ def _compute_pending(self):
         for picking in self:
             if picking.can_handle_partials() is False and picking.u_pending is True:
-                raise UserError(_("Cannot validate %s until all of its preceeding pickings are done.") % picking.name)
+                raise UserError(_("Cannot validate %s until all of its preceding pickings are done.") % picking.name)
```
Bit long :P

ws-mps committed:
[918] Cut long lines as per CR
